pkg/cloud/services/ec2/bastion.go

```diff
@@ -82,6 +82,7 @@ func (s *Service) ReconcileBastion() error {
 			record.Warnf(s.scope.InfraCluster(), "FailedFetchingBastion", "Failed to fetch default bastion instance: %v", err)
 			return err
 		}
+		// TODO(mtulio): add support of BYO IP for bastion
```

Comment:
does this still need to be addressed?

Reply:
it seems we can drop this for that version. I can't see the machine field `PublicIP` being used in the bastion workflow.
It seems Bastion will always get public subnets, which has map IP on launch by default on CAPA.
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/4c41955afcb2e2bce07af1754e4c34226c07debe/pkg/cloud/services/ec2/bastion.go#L181